File: topcoder/SRM363-DIV2/250_MirroredClock.py
class MirroredClock:
	def whatTimeIsIt(self, time):
		mm = (60 - int(time[3:])) % 60
		hh = (12 - int(mm>0) - int(time[:2])) % 12
		return "%02d:%02d" % (hh, mm)

File: topcoder/SRM300~399/SRM363-DIV2/250_MirroredClock.cpp
#include <iostream>
#include <string>
using namespace std;

class MirroredClock{
public:
	string whatTimeIsIt(string time){
		string mm = to_string((60 - stoi(time.substr(3, 2))) % 60);
		string hh = to_string(((mm == "0") ? 12 - stoi(time.substr(0, 2)) : 11 - stoi(time.substr(0, 2))) % 12);
		if(mm.size() == 1) mm = "0" + mm;
		if(hh.size() == 1) hh = "0" + hh;
		return hh + ":" + mm;
	}
};